assets/pages/new-contributions-landing/contributionsLandingReducer.js

```diff
@@ -2,47 +2,66 @@
 
 // ----- Imports ----- //
 
-import { combineReducers, type CombinedReducer } from 'redux';
-
-import { contributionSelectionReducerFor } from 'components/contributionSelection/contributionSelectionReducer';
-import { payPalContributionButtonReducerFor } from 'components/payPalContributionButton/payPalContributionButtonReducer';
-
-import type { State as ContributionSelectionState } from 'components/contributionSelection/contributionSelectionReducer';
-import type { Action as ContributionSelectionAction } from 'components/contributionSelection/contributionSelectionActions';
-import type { State as PayPalState } from 'components/payPalContributionButton/payPalContributionButtonReducer';
-import type { Action as PayPalAction } from 'components/payPalContributionButton/payPalContributionButtonActions';
-
-import type { CommonState } from 'helpers/page/page';
-import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
+import { type PaymentMethod } from 'helpers/checkouts';
+import { amounts, type Amount, type Contrib } from 'helpers/contributions';
+import { type CommonState } from 'helpers/page/page';
+import { type CountryGroupId } from 'helpers/internationalisation/countryGroup';
 
+import { type Action } from './contributionsLandingActions';
 
 // ----- Types ----- //
 
 type PageState = {
-  selection: ContributionSelectionState,
-  payPal: PayPalState,
+  contributionType: Contrib,
+  paymentMethod: PaymentMethod,
+  amount: Amount | null,
+  otherAmount?: number,
 };
 
 export type State = {
   common: CommonState,
   page: PageState,
 };
 
-type PageReducer = CombinedReducer<PageState, ContributionSelectionAction | PayPalAction>;
-
+// ----- Initial state ----- //
 
 // ----- Functions ----- //
 
-function createPageReducerFor(countryGroupId: CountryGroupId): PageReducer {
-
-  return combineReducers({
-    selection: contributionSelectionReducerFor('CONTRIBUTE_SECTION', countryGroupId),
-    payPal: payPalContributionButtonReducerFor('CONTRIBUTE_SECTION'),
-  });
-
+function initReducer(countryGroupId: CountryGroupId) {
+
+  const initialState: PageState = {
```

Comment:
In the `PageState` type above you have an optional `otherAmount` field but no `showOtherAmount` boolean. I guess it's Flow's [width subtyping](https://flow.org/en/docs/lang/width-subtyping/) plus `otherAmount` being optional that's allowing this to type-check but would it be better if these types matched up? Maybe just check truthiness of `otherAmount` in [`mapStateToProps`](https://github.com/guardian/support-frontend/blob/4628f3c96482e74b289b2b1f807646e7a8a3e701/assets/pages/new-contributions-landing/components/ContributionAmount.jsx#L32-L36) rather than having a specific boolean

Reply:
Fixed, good spot!